File: 876/main.py
from typing import Optional

class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next
        
def printLinkedList(list: ListNode):
    printed = list
    printedArr = []
    while printed :
        printedArr.append(printed.val)
        printed = printed.next
    print(printedArr)
        


def middleNode(head: Optional[ListNode]) -> Optional[ListNode]:
    pointerA = pointerB = head

    while pointerB and pointerB.next :
        pointerA = pointerA.next
        pointerB = pointerB.next.next

    return pointerA




questionParam = [2,3,4,5,6]
linkedListParam = ListNode(1)
paramHead = linkedListParam

for param in questionParam:
    linkedListParam.next = ListNode(param)
    linkedListParam = linkedListParam.next

printLinkedList(paramHead)

printLinkedList(middleNode(paramHead))

File: 876/main.go
package main

import (
	"fmt"
)

type ListNode struct {
	Val  int
	Next *ListNode
}

func middleNode(head *ListNode) *ListNode {
	slow, fast := head, head

	for fast != nil && fast.Next != nil {
		fast = fast.Next.Next
		slow = slow.Next
	}
	return slow
}

func printLinkedList(head *ListNode) {
	printed := head
	printedArr := []int{}

	for printed != nil {
		printedArr = append(printedArr, printed.Val)
		printed = printed.Next
	}

	fmt.Println(printedArr)
}

func main() {
	questionParam := []int{2, 3, 4, 5, 6}

	paramHead := &ListNode{Val: 1}
	linkedListParam := paramHead

	for _, param := range questionParam {
		linkedListParam.Next = &ListNode{Val: param}
		linkedListParam = linkedListParam.Next
	}

	printLinkedList(paramHead)
	printLinkedList(middleNode(paramHead))
}
